calcom/cal.com
# feat: min max chars for long text

/claim #15577 
## What does this PR do?
This PR adds an optional minimum and maximum character limit to long text questions in the booking form.




- Fixes #15577  (GitHub issue number)
- Fixes CAL-15577 (Linear issue number - should be visible at the bottom of the GitHub issue description)


https://www.loom.com/share/a244c881d5214b49b6b25c7f52467f2e

## Mandatory Tasks (DO NOT REMOVE)


## How should this be tested?


- Are there environment variables that should be set?
  N/A
- What is expected (happy path) to have (input and output)?

1. Click on Event Types in the sidebar.
2. Click on an event.
3. Inside the event, click on the Advanced tab on the left sidebar.
4. Add / edit a question.
5. Set type to Long text
6. Set the min &/or max characters.
7. Save the question.
8. Save the form.
9. Click on the Preview button in the event header.
10. Select a date on the calendar.
11. While filling out the form, click the save button with:
a. Min &/or max character limit unsatisfied -> should show red text with an error.
b. All the conditions satisfied -> Should save successfully.

## Checklist



- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my changes generate no new warnings
